### ISSUE TITLE
pyright behavior is different from mypy when it  checks union type which has type variable with bound

### ISSUE THREAD

**Issue by kentac55** [microsoft/pyright#618]
**Describe the bug**

pyright behavior is different from mypy when it  checks union type which has type variable with bound

**To Reproduce**

write code as below:

```python
from typing import Union, Generic, TypeVar

class InnerA:
    pass

class InnerB:
    pass

T = TypeVar("T", bound=InnerA)

class A(Generic[T]):
    pass

class B:
    pass

U = Union[A[T], B]

a: U[InnerA]
b: U[InnerB]
```

check with mypy and pyright:

```sh
$ mypy pyt/a.py
pyt/a.py:19: error: Type argument "pyt.a.InnerB" of "A" must be a subtype of "pyt.a.InnerA"

$ pyright pyt/a.py
typingsPath /home/kc5m/src/localhost/kc5m/pyt/typings is not a valid directory.
Searching for source files
Found 1 source file
0 errors, 0 warnings
Completed in 0.864sec
```

**Expected behavior**

pyright should report error

**Screenshots or Code**

**VS Code extension or command-line**

```sh
$ pyright --version
pyright 1.1.31

$ mypy --version
mypy 0.770
```

**Additional context**

**Comment by erictraut** [comment]
Thanks for the bug report. This is indeed a bug. I've introduced a fix, and it will be in the next published version of pyright.

I also found and fixed a related bug. Pyright wasn't reporting an error if the code provided too many type arguments for a generic union type alias. For example: `c: U[InnerA, int]`.

**Comment by erictraut** [comment]
This is now fixed in version 1.1.32, which I just published.
